### DISCUSSION TITLE
Generation of two dimensions with large differences（Al-7si casting alloy）

### DISCUSSION THREAD

**Discussion by yqy19921021** [neperfepx/neper#428] [Q&A] [answered]
Hello 
I hope to generate the 3D structure of Al-Si casting alloy. The image is as follows. It looks very simple, but I have been on this issue for a long time, still couldn’t found the right strategy.So ask for help.
The main problem is that the number of main phase grains is very small, while the the second phase‘s size distribution is banded (because it is eutectic structure).
To put it simply, dozens of spherical second phases are distributed in the grain boundaries of large grains with a number of less than 5.
I also have another simple question. What is the definition of weight? It's definition is, as the document says—width for a lamellar tessellation?

Thanks!
Best regards.
![2](https://user-images.githubusercontent.com/74890058/162025202-205535b7-3525-46df-8950-ca51f569d635.png)
![3](https://user-images.githubusercontent.com/74890058/162025215-4117e405-0587-4e5b-a337-c30d51e6585e.png)
![1](https://user-images.githubusercontent.com/74890058/162025217-3160ee35-23c5-486e-a109-7e11eaad16bb.png)

**Comment by rquey** [comment, marked as answer]
This is a complex structure... My best guess would be to generate a standard polycrystal, mesh it, and select elements at appropriate locations to represent the second phase.

For the definition of the weight, see https://neper.info/papers.html#id2.

**Reply by mkasemer** [reply]
Alternatively, depending on how you will use this tessellation, you could employ a multilevel tessellation:
- First level, 3-4 cells
- Second level, 100s of subcells per parent cell

You could then select (outside of Neper, using a custom script) the majority of subcells to be the same orientation, and reserve a certain fraction to be the minor "phase". If your program will make no distinction between the subcells that have the same orientation / designation, this should be fine...

**EDIT** Choosing subcells to be minor phase at random would likely bias toward subcells that are interior to the parent cells. If you need more along the parent cell boundaries, you could selectively choose. This may, of course, lead to a much finer mesh than necessary, but if the premium is on the morphology...

**Reply by yqy19921021** [reply]
Thank you for your reply！ 
I’ll try！

**Reply by yqy19921021** [reply]
Perhaps further simplification,consider the second phase as pore（the second phase is brittle phase, so I think it is a reasonable simplification）.
Therefore, the structure becomes： unevenly distributed holes on grain boundaries of large grains.
Does that make the model easier to build？
Thank you!!!

**Reply by rquey** [reply]
No, not really. This is probably not very difficult to do, but it is not available.
